lua-api-crates/table-funcs/src/lib.rs

```diff
@@ -22,27 +23,30 @@ pub fn register(lua: &Lua) -> anyhow::Result<()> {
 
 #[derive(Debug, Clone, PartialEq, Eq)]
 enum ConflictMode {
-   Keep,
-   Force,
-   Error,
+    Keep,
+    Force,
+    Error,
 }
 
 impl<'lua> mlua::FromLua<'lua> for ConflictMode {
     fn from_lua(value: LuaValue<'lua>, _: &'lua Lua) -> mlua::Result<Self> {
         match value {
-            LuaValue::String(s) => {
-                match s.to_str() {
-                    Ok("Keep") => Ok(ConflictMode::Keep),
-                    Ok("keep") => Ok(ConflictMode::Keep),
-                    Ok("Force") => Ok(ConflictMode::Force),
-                    Ok("force") => Ok(ConflictMode::Force),
-                    Ok("Error") => Ok(ConflictMode::Error),
-                    Ok("error") => Ok(ConflictMode::Error),
-                    _ => Err(mlua::Error::runtime("Unknown string. Expected 'Keep', 'Force' or 'Error'".to_string()))
-                }
-            }
+            LuaValue::String(s) => match s.to_str() {
+                Ok("Keep") => Ok(ConflictMode::Keep),
+                Ok("keep") => Ok(ConflictMode::Keep),
+                Ok("Force") => Ok(ConflictMode::Force),
+                Ok("force") => Ok(ConflictMode::Force),
+                Ok("Error") => Ok(ConflictMode::Error),
+                Ok("error") => Ok(ConflictMode::Error),
+                _ => Err(mlua::Error::runtime(
+                    "Unknown string. Expected 'Keep', 'Force' or 'Error'".to_string(),
+                )),
+            },
             LuaValue::Error(err) => Err(err),
-            other => Err(mlua::Error::runtime(format!("Expected a Lua string. Got something of type: {}", other.type_name()))),
+            other => Err(mlua::Error::runtime(format!(
+                "Expected a Lua string. Got something of type: {}",
+                other.type_name()
+            ))),
```

Comment:
This should all be derivable via FromDynamic and ToDynamic. Please share the errors you mentioned when you tried my original suggestion

Reply:
```rust
error[E0277]: the trait bound `ConflictMode: wezterm_dynamic::todynamic::ToDynamic` is not satisfied
  --> lua-api-crates/table-funcs/src/lib.rs:30:1
   |
30 | impl_lua_conversion_dynamic!(ConflictMode);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | the trait `wezterm_dynamic::todynamic::ToDynamic` is not implemented for `ConflictMode`
   | required by a bound introduced by this call
   |
   = help: the following other types implement trait `wezterm_dynamic::todynamic::ToDynamic`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             usize
           and 152 others
note: required by a bound in `to_lua`
  --> /Users/daniel/personal/wezterm/luahelper/src/lib.rs:12:24
   |
12 | pub fn to_lua<'lua, T: ToDynamic>(
   |                        ^^^^^^^^^ required by this bound in `to_lua`
   = note: this error originates in the macro `impl_lua_conversion_dynamic` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `ConflictMode: wezterm_dynamic::fromdynamic::FromDynamic` is not satisfied
  --> lua-api-crates/table-funcs/src/lib.rs:30:1
   |
30 | impl_lua_conversion_dynamic!(ConflictMode);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `wezterm_dynamic::fromdynamic::FromDynamic` is not implemented for `ConflictMode`
   |
   = help: the following other types implement trait `wezterm_dynamic::fromdynamic::FromDynamic`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             usize
           and 152 others
note: required by a bound in `luahelper::from_lua`
  --> /Users/daniel/personal/wezterm/luahelper/src/lib.rs:20:26
   |
20 | pub fn from_lua<'lua, T: FromDynamic>(value: mlua::Value<'lua>) -> Result<T, mlua::Error> {
   |                          ^^^^^^^^^^^ required by this bound in `from_lua`
   = note: this error originates in the macro `impl_lua_conversion_dynamic` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `table-funcs` (lib) due to 2 previous errors
```

Reply:
Wait, I made a mistake. I will update the above.

Reply:
I forgot to add `FromDynamic` and `ToDynamic` to the `derive` in the above log. Here is the log with those added:

```rust
error: cannot find derive macro `FromDynamic` in this scope
  --> lua-api-crates/table-funcs/src/lib.rs:24:17
   |
24 | #[derive(Debug, FromDynamic, ToDynamic, Clone, PartialEq, Eq)]
   |                 ^^^^^^^^^^^

error: cannot find derive macro `ToDynamic` in this scope
  --> lua-api-crates/table-funcs/src/lib.rs:24:30
   |
24 | #[derive(Debug, FromDynamic, ToDynamic, Clone, PartialEq, Eq)]
   |                              ^^^^^^^^^

error[E0277]: the trait bound `ConflictMode: wezterm_dynamic::todynamic::ToDynamic` is not satisfied
  --> lua-api-crates/table-funcs/src/lib.rs:30:1
   |
30 | impl_lua_conversion_dynamic!(ConflictMode);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | the trait `wezterm_dynamic::todynamic::ToDynamic` is not implemented for `ConflictMode`
   | required by a bound introduced by this call
   |
   = help: the following other types implement trait `wezterm_dynamic::todynamic::ToDynamic`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             usize
           and 152 others
note: required by a bound in `to_lua`
  --> /Users/daniel/personal/wezterm/luahelper/src/lib.rs:12:24
   |
12 | pub fn to_lua<'lua, T: ToDynamic>(
   |                        ^^^^^^^^^ required by this bound in `to_lua`
   = note: this error originates in the macro `impl_lua_conversion_dynamic` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `ConflictMode: wezterm_dynamic::fromdynamic::FromDynamic` is not satisfied
  --> lua-api-crates/table-funcs/src/lib.rs:30:1
   |
30 | impl_lua_conversion_dynamic!(ConflictMode);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `wezterm_dynamic::fromdynamic::FromDynamic` is not implemented for `ConflictMode`
   |
   = help: the following other types implement trait `wezterm_dynamic::fromdynamic::FromDynamic`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             usize
           and 152 others
note: required by a bound in `luahelper::from_lua`
  --> /Users/daniel/personal/wezterm/luahelper/src/lib.rs:20:26
   |
20 | pub fn from_lua<'lua, T: FromDynamic>(value: mlua::Value<'lua>) -> Result<T, mlua::Error> {
   |                          ^^^^^^^^^^^ required by this bound in `from_lua`
   = note: this error originates in the macro `impl_lua_conversion_dynamic` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `table-funcs` (lib) due to 4 previous errors
```

Reply:
I think you just need to `use wezterm_dynamic::{FromDynamic, ToDynamic};` at the top of the file?

Reply:
> I think you just need to `use wezterm_dynamic::{FromDynamic, ToDynamic};` at the top of the file?

Ah, I see what went wrong. I needed to update the `Cargo.toml` and add this. It works now (though it won't accept e.g. `keep` now, only `Keep`).